processor/src/main/java/com/linecorp/decaton/processor/runtime/PartitionContext.java

```diff
@@ -52,7 +52,7 @@ public PartitionContext(PartitionScope scope, Processors<?> processors, int maxP
         metrics = Metrics.withTags("subscription", scope.subscriptionId(),
                                    "topic", tp.topic(),
                                    "partition", String.valueOf(tp.partition()))
-                .new PartitionStateMetrics();
+                .new PartitionStateMetrics(commitControl::pendingOffsetsCount, () -> paused() ? 1 : 0);
```

Comment:
I remember there was a discussion about letting metrics to call this `pendingOffsetsCount` directly with concern of race between consumer thread and metric scrape.
After I looked into its implementation, what it does is just calling .size() against ArrayDeque which should return ultimately fast and since other synchronized methods call by consumer thread are just tiny part of entire consuming-loop I thought it would be safe here. Different thoughts? @ocadaruma

Reply:
Sounds completely fine.